tests/integration/application/applicationapi_test.go

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package application
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/suite"
+	"net/http"
+	"testing"
+)
+
+const (
+	testServerURL = "https://localhost:8095"
+)
+
+var (
+	preCreatedApp = Application{
+		ID:                  "550e8400-e29b-41d4-a716-446655440000",
+		Name:                "Test SPA",
+		Description:         "Initial testing App",
+		ClientID:            "client123",
+		CallbackURL:         []string{"http://example.com/callback"},
+		SupportedGrantTypes: []string{"client_credentials"},
+	}
+
+	appToCreate = Application{
+		Name:                "My App",
+		Description:         "A demo application",
+		ClientID:            "abc1237",
+		ClientSecret:        "s3cret",
+		CallbackURL:         []string{"http://localhost/callback"},
+		SupportedGrantTypes: []string{"authorization_code", "client_credentials"},
+	}
+
+	appToUpdate = Application{
+		Name:                "Updated App",
+		Description:         "Updated Description",
+		ClientID:            "Updated abc1237",
+		ClientSecret:        "Updated s3cret",
+		CallbackURL:         []string{"http://localhost/callback2"},
+		SupportedGrantTypes: []string{"authorization_code2", "client_credentials2"},
+	}
+)
+
+var createdAppID string
+
+type ApplicationAPITestSuite struct {
+	suite.Suite
+}
+
+func TestApplicationAPITestSuite(t *testing.T) {
+
+	suite.Run(t, new(ApplicationAPITestSuite))
+}
+
+// SetupSuite test application creation
+func (ts *ApplicationAPITestSuite) SetupSuite() {
+	id, err := createApplication(ts)
```

Comment:
```suggestion

	id, err := createApplication(ts)
```

Let's have an empty line after the function definition to improve clarity